packages/sanity/src/core/preview/utils/optimizeQuery.ts

```diff
@@ -52,13 +52,13 @@ export function toQuery(combinedSelections: CombinedSelection[]): string {
 export function reassemble(
   queryResult: Result[],
   combinedSelections: CombinedSelection[],
-): (Doc | null)[] {
-  return queryResult.reduce((reprojected: (Doc | null)[], subResult, index) => {
+): (Doc | undefined)[] {
+  return queryResult.reduce((reprojected: (Doc | undefined)[], subResult, index) => {
     const map = combinedSelections[index].map
     map.forEach((resultIdx, i) => {
       const id = combinedSelections[index].ids[i]
       const found = subResult.find((doc) => doc._id === id)
-      reprojected[resultIdx] = found || null
+      reprojected[resultIdx] = found ?? undefined
```

Comment:
This is what fixes the issue – returning `null` here caused the check here to always be `false`, thus never invoking the "slow fetch": https://github.com/sanity-io/sanity/blob/b77f684f04c4d090fafb94e1d43184526c461eba/packages/sanity/src/core/preview/observeFields.ts#L107